addons/account/models/account_move_line.py

```diff
@@ -2033,6 +2033,20 @@ def _prepare_reconciliation_single_partial(self, debit_values, credit_values, sh
         debit_fully_matched = compare_amounts <= 0
         credit_fully_matched = compare_amounts >= 0
 
+        def get_amount_range_after_rate(currency_from, currency_to, amount, rate):
+            # Suppose balance=1000, rate=12.
+            # 1000.0 could be the result of a rounding of [999.995, 1000.0049999999999].
+            # Let's say the target currency could be [999.995 * 12, 1000.005 * 12] = [11999.94, 12000.06]
+            # instead of just 120000.
+            if not rate:
+                return (0.0, 0.0)
+            sub_rounding = currency_from.rounding / 10
+            return (
+                currency_to.round((amount - (5 * sub_rounding)) * rate),
```

Comment:
Why not just `/ 2` instead of `/ 10 * 5`
```suggestion
            half_rounding = currency_from.rounding / 2
            return (
                currency_to.round((amount - half_rounding) * rate),
```

Reply:
@svfu-odoo Sure :rofl: I had a mental bug this day^^